### PR TITLE
Tiles blessed by holy water now block servant warp-in/Eminence snooping

### PR DESCRIPTION
**Pull request by ghost**
:cl: Xhuis
balance: Floors blessed with holy water prevent servants from warping in and the Eminence from crossing them.
balance: The Eminence can never enter the Chapel.
/:cl:

Right now, the Eminence has no real counterplay to being all-seeing; even the AI has camera restrictions. This gives a way to prevent the Eminence from snooping in an area by covering it with holy water, meaning you can secure high-security areas like the Brig from invisible spying. It also prevents servants warping in, both for consistency and so that sensitive areas vulnerable to silent intrusion can be secured without constant guarding. This weakens warcult considerably if the crew are diligent, since it nullifies the Eminence's ability to understand speech if the entire station is covered in holy water.

Since tiles can just be pried up to remove their protection, servants themselves can either create (by prying up floors or welding down walls) or locate unshielded tiles, meaning that usually without exception they can still get onto the station with no trouble, since it's so large that covering *every* tile would be very difficult. If it proves problematic, I can implement some way to unbless a tile from Reebe to prevent servants from being unable to get to the station at all.

Point of note: the Eminence can never enter the Chapel, even if it isn't blessed. This means that even if servants can secure and unbless the entire station, the Chapel will always remain mostly blind to them without camera visibility or an inside man.

This also makes clockwork floors immune to holy water blessing, because having otherwise is silly.

### CODE REVIEW

**Comment by Bawhoppen** [comment]
Holy water giving a tint is a lame hugbox thing

**Comment by Xhuis** [comment]
What does that even mean?

**Comment by Bawhoppen** [comment]
It is more interesting if it's a hidden effect you don't necessarily expect, also making the game overall more challenging

**Comment by Bawhoppen** [comment]
I know that looks like a fwooshpost, but I hope you understand my point

**Comment by AdamElTablawy** [comment]
It is also as useless as a fwooshpost is. The visibility is basically necessary here, as unlike jaunting, all this will do is cause minor communication issues in not knowing where a warp down is actually possible vs the wizard getting cucked by the holy tiles they couldn't see.

QoL.

**Comment by Bawhoppen** [comment]
QoL is literally the non-argument of the millenium

**Comment by XDTM** [comment]
It helps the cult giving feedback on blocked areas, but also the crew as it will highlight sabutage; currently either you regularly spray the areas you want to protect, or you'll never be sure someone didn't rip the floor off to remove it.

**Comment by evsey9** [comment]
two words:
smoke. machines.

**Comment by ChangelingRain** [comment]
Blessed tiles should be separate from tiles with NOJAUNT, since the xenobio blueprints can override the color change.

**Comment by WJohn** [comment]
Water being invisible for 90% of their duration except for a brief flash was "fun and unexpected" until we all realized it was *FUCKING STUPID*, bawhoppen. Don't just make things which do not need to be invisible, invisible.

**Comment by uraniummeltdown** [comment]
Good change, invisible things that affect gameplay suck
I'd suggest making a "holy" shiny/bright/hallelujah/blessed overlay over blessed floors, it would fix issues like the xenobio blueprints etc

**Comment by Potato-Masher** [comment]
I'm wondering what would happen if the Chaplain was the only one to see this effect. If that seems like putting too much focus on one person, how about having everyone else see pulses of the tint effect faintly, but the chaplain sees it all of the time.

But, I do like the idea of this change. Blessed tiles felt kinda pointless feature wise as they were very situational. Not only does this fix that, but now this should deter people from war cult in all the time.

I would also like to add that a Chemist with a tank of holy water is going to be a scary person for the cult.

**Comment by Xhuis** [comment]
I'm not really sure what a "blessed" overlay would look like to distinguish it from other colors tiles might impose. I suppose I could think on it.

**Comment by Bawhoppen** [comment]
@WJohn I am aware this change isn't going to be the one that would ruin the game, but every time we take away a unique and interesting feature (relatively), for just another cheap mechanic the game is overall worse. It's a path we should not go down.

**Comment by Xhuis** [comment]
Making blessed tiles have a slight color isn't taking away a unique and interesting feature, what the heck are you smoking?

**Comment by Bawhoppen** [comment]
It's certainly more interesting than just being another typical visible obstacle. If you don't understand what I mean at this point, I don't think there's a better explanation.

**Comment by Bawhoppen** [comment]
I'm just saying you're making it more generic. Of course I'm not acting like this is an absolute critical thing, but it's just a lame change.

File: code/game/gamemodes/clock_cult/clock_mobs/_eminence.dm [hunk 1]
```diff
@@ -20,7 +20,16 @@
 /mob/camera/eminence/Move(NewLoc, direct)
 	var/OldLoc = loc
 	if(NewLoc && !istype(NewLoc, /turf/open/indestructible/reebe_void))
-		forceMove(get_turf(NewLoc))
+		var/turf/T = get_turf(NewLoc)
+		if(T.flags_1 & NOJAUNT_1)
+			T.visible_message("<span class='warning'>[T] suddenly emits a ringing sound!</span>", ignore_mob = src)
```
Comment (ninjanomnom) [inline code/game/gamemodes/clock_cult/clock_mobs/_eminence.dm]: This seems like it can easily get spammy, maybe stun the eminence?

**Review by ninjanomnom — CHANGES_REQUESTED** [review]

**Comment by coiax** [comment]
Do what @ChangelingRain says, make a difference between blessing the tile, and making a tile non-jaunt for technical reasons (see transit space)

**Comment by ninjanomnom** [comment]
That'd be a different pr and I've got an idea that may be better than whatever is done here @coiax

**Comment by coiax** [comment]
@Xhuis see how I did it in #33374, which I was mostly waiting on 1) sprites 2) being able to limit showing the blessing object to chaplains, cultists and wizards

**Comment by Xhuis** [comment]
I'll cut out the color change in favor of that PR, then. Thanks for messing with it!

File: code/game/gamemodes/clock_cult/clock_mobs/_eminence.dm (hunk 1 above)
Comment (Xhuis) [reply code/game/gamemodes/clock_cult/clock_mobs/_eminence.dm]: #33605 will make it a non-issue.

Comment (ninjanomnom) [reply code/game/gamemodes/clock_cult/clock_mobs/_eminence.dm]: not necessarily, any intervening chat will make it appear again and the sound plays again as well

**Comment by coiax** [comment]
Also, @Xhuis the Eminence is sent to random places on the station, does this check for holyness as well?

File: code/game/gamemodes/clock_cult/clock_mobs/_eminence.dm [hunk 2]
```diff
@@ -20,7 +21,18 @@
...
+		if(T.flags_1 & NOJAUNT_1)
+			if(last_failed_turf != T)
+				T.visible_message("<span class='warning'>[T] suddenly emits a ringing sound!</span>", ignore_mob = src)
+				playsound(T, 'sound/machines/clockcult/ark_damage.ogg', 75, FALSE)
+				last_failed_turf = T
+			to_chat(src, "<span class='warning'>This turf is consecrated and can't be crossed!</span>")
+			return
+		if(istype(get_area(T), /area/chapel))
+			to_chat(src, "<span class='warning'>The Chapel is hallowed ground under a heretical deity, and can't be accessed!</span>")
+			return
```
Comment (coiax) [inline code/game/gamemodes/clock_cult/clock_mobs/_eminence.dm]: When I finish my blessing changes, the chapel will just start with blessing objects in its area.

**Comment by Xhuis** [comment]
It doesn't check for holiness at the moment, nope.

File: code/game/gamemodes/clock_cult/clock_mobs/_eminence.dm (hunk 2 above)
Comment (Xhuis) [reply code/game/gamemodes/clock_cult/clock_mobs/_eminence.dm]: Oh, I see.

**Review by ninjanomnom — APPROVED** [review]

**Comment by DaxDupont** [comment]
and @coiax never finished his blessed titles view

**Pull request merged**